urld0f681c558_farom57_AstroidDriver_tgz-pJ8-laazotea_indi_server_examples_MinimumINDIServerJ8/src/laazotea/indi/INDISexagesimalFormatter.java

```diff
@@ -0,0 +1,355 @@
+      fractionLength = Integer.parseInt(frLength);
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Illegal sexagesimal length or fraction length");
+    }
+
+    if ((fractionLength != 3) && (fractionLength != 5) && (fractionLength != 6) && (fractionLength != 8) && (fractionLength != 9)) {
+      throw new IllegalArgumentException("Illegal sexagesimal fraction length");
+    }
+  }
+
+  /**
+   * Parses a sexagesimal number. DO NOT USE IT. THIS IS A PRELIMINARY VERSION
+   * AND DOES NOT WORK AS EXPECTED. THIS METHOD WILL DISAPEAR IN FUTURE VERSIONS
+   * OF THE CLASS.
+   *
+   * @param number NOT USED
+   * @return NOT USED
+   * @throws IllegalArgumentException
+   * @deprecated
+   */
+  @Deprecated
+  public double parseSexagesimal2(String number) throws IllegalArgumentException {
+    number = number.replace(' ', ':');
+    number = number.replace(';', ':');
+
+//    System.out.println(" ->" + number + " ; " + format);
+//    System.out.flush();
+
+    if (number.indexOf(":") == -1) {  // If there are no separators maybe they have sent just a single double
+      try {
+        double n = Double.parseDouble(number);
+
+        return n;
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("Sexagesimal number format not correct (not even single number)");
+      }
+    }
+
+    StringTokenizer st = new StringTokenizer(number, ":", false);
+
+    double degrees = 0;
+    double minutes = 0;
+    double seconds = 0;
+
+    try {
+      String aux = st.nextToken().trim();
+
+      if (aux.length() > 0) {
+        degrees = Double.parseDouble(aux);
+      }
+
+      aux = st.nextToken().trim();
+
+      if (aux.length() > 0) {
+        minutes = Double.parseDouble(aux);
+      }
+
+      if (fractionLength > 5) {
+        aux = st.nextToken().trim();
+
+        if (aux.length() > 0) {
+          seconds = Double.parseDouble(aux);
+        }
+      }
+
+    } catch (NoSuchElementException e) {
+      throw new IllegalArgumentException("Sexagesimal number format not correct");
+    } catch (NumberFormatException e) {
+      throw new IllegalArgumentException("Sexagesimal number component not correct");
+    }
+
+    double res = degrees;
+    if (degrees > 0) {
+      res += (minutes / 60.0) + (seconds / 3600.0);
+    } else {
+      res -= (minutes / 60.0) + (seconds / 3600.0);
+    }
+
+    return res;
+  }
+
+  /**
+   * Parses a sexagesimal number. The input
+   * <code>String</code> is formatted as a maximum of three doubles separated by
+   * : ; or a blank space. The first number represents the number of degrees,
+   * the second is the number of minutes and the third is the number of seconds.
+   *
+   * @param number The number to be parsed.
+   * @return The parsed double.
+   * @throws IllegalArgumentException if the number format is not correct.
+   */
+  public double parseSexagesimal(String number) throws IllegalArgumentException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 241-248 into a separate method.